### PR TITLE
Add virtual hooks to enable parallel frontends in opm-simulators

### PR DESCRIPTION
**Pull request by akva2**

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this opm-material=391 opm-simulators=2311 please

File: opm/parser/eclipse/EclipseState/EclipseState.hpp [hunk 1]
```diff
@@ -98,7 +98,10 @@ namespace Opm {
         const EDITNNC& getInputEDITNNC() const;
         bool hasInputEDITNNC() const;
 
+        // The potentially parallelized field properties
         virtual const FieldPropsManager& fieldProps() const;
+        // Always the non-parallel field properties
+        virtual const FieldPropsManager& fullFieldProps() const;
```
Comment (joakim-hove) [inline opm/parser/eclipse/EclipseState/EclipseState.hpp]: Can this be `global`instead of `full`?

**Comment by joakim-hove** [comment]
If the default constructed `EclipseState`or `EclispeGrid` are actually used I assume things will fail spectacularly - is it possible handle that somewhat gracefully - without dragging along a `bool is_valid`?

File: opm/parser/eclipse/EclipseState/EclipseState.hpp (hunk 1 above)
Comment (akva2) [reply opm/parser/eclipse/EclipseState/EclipseState.hpp]: sure.

**Comment by akva2** [comment]
the eclipse state accesses are already protected by asserts. i can do the same for the eclipsegrid if you want (ie virtualize and override in the parallel eclipse state class).

**Comment by akva2** [comment]
jenkins build this opm-material=391 opm-simulators=2311 please

**Comment by akva2** [comment]
jenkins build this opm-material=391 opm-simulators=2311 please

**Comment by akva2** [comment]
jenkins run norne_parallel opm-simulators=2311 please

**Comment by akva2** [comment]
jenkins run norne_parallel opm-simulators=2311 please

**Comment by akva2** [comment]
alright, norne also passes.

**Comment by akva2** [comment]
jenkins run norne_parallel opm-simulators=2311 please

**Comment by akva2** [comment]
green, approved. merging.

**Pull request merged**